### DISCUSSION TITLE
Is there a way to set/increase timeouts for detection.detect(state.data) methods?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [exadel-inc/CompreFace#839] [Q&A]
Is there a way to set/increase timeouts for detection.detect(state.data) methods?

**Comment by pospielov** [comment]
The default timeout is set by nginx and it's now 60 seconds.
You can set this timeout in nginx configuration and replace it using docker volume:
https://github.com/exadel-inc/CompreFace/blob/master/ui/nginx/templates/nginx.conf.template

**Reply by martinenkoEduard** [reply]
In this nginx.conf.template example there is 10 seconds..

**Reply by martinenkoEduard** [reply]
What file should I replace? 
Is it defined only by nginx ? 
Aren't there any related Java Spring?

**Reply by pospielov** [reply]
10s is not the timeout you need.
The default timeout is the 60s.
You need [proxy_read_timeout](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_read_timeout):
@VolodymyrBushko could you look at our Java timeouts?

**Reply by VolodymyrBushko** [reply]
There is 60 seconds read-timeout on the java side and 5 retries if something went wrong. You can't customize that, but we will soon add the ability to change the default settings.

**Reply by VolodymyrBushko** [reply]
We have added the ability to adjust connection and read timeout on the Java side, but you should wait for version 1.1 to be released or build it on your own from the 'master' branch.

There is a .env file through which you can adjust timeout settings.

connection_timeout - is used  to set the connection timeout (default is 10 seconds)

read_timeout - is used to set the read timeout (default is 60 seconds)
